components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/throttling/ThrottleHandler.java

```diff
@@ -275,43 +275,42 @@ private boolean doRoleBasedAccessThrottlingWithCEP(MessageContext synCtx, Config
                                 APIConstants.METRICS_PREFIX, this.getClass().getSimpleName(), RESOURCE_THROTTLE));
                         Timer.Context context1 = timer1.start();
 
-                        if (conditionGroupDTOs != null && conditionGroupDTOs.length > 0) {
-                            // Checking Applicability of Conditions is a relatively expensive operation. So we are
-                            // going to check it only if the API/Resource is throttled out.
-                            if (getThrottleDataHolder().isAPIThrottled
-                                    (resourceLevelThrottleKey)) {
-                                if (getThrottleDataHolder().isConditionsAvailable(resourceLevelThrottleKey)) {
-                                    Map<String, List<ConditionDto>> conditionDtoMap = getThrottleDataHolder()
-                                            .getConditionDtoMap(resourceLevelThrottleKey);
+                        if (getThrottleDataHolder().isAPIThrottled
+                                (resourceLevelThrottleKey)) {
```

Comment:
can't this go in the same line 
`if (getThrottleDataHolder().isAPIThrottled (resourceLevelThrottleKey)) {`

Reply:
Fixed in https://github.com/wso2/carbon-apimgt/pull/9250